solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,338 @@
+        }
+
+        private Object readFromFile(File file) {
+            Object result = null;
+
+            try (FileInputStream fin = new FileInputStream(file);
+                 ObjectInputStream oin = new ObjectInputStream(fin)) {
+
+                    result = oin.readObject();
+
+            } catch (ClassNotFoundException | IOException e) {
+                throw new CacheProxyException("Error occurred while loading file: " + file.getName(), e);
+            }
+            return result;
+        }
+
+        private Object readZipFile(File file) {
+            Object result = null;
+            try {
+                File zipFileToGet = new File(file.getAbsolutePath());
+
+                if (zipFileToGet.exists()) {
+                    try (ZipFile zipFile = new ZipFile(file.getAbsolutePath())) {
+                        ZipEntry zipEntry = zipFile.getEntry(
+                                file.getName().replace(".zip", ""));
+
+                        try (InputStream in = zipFile.getInputStream(zipEntry)) {
+                            byte[] bytes = getBytes(in);
+
+                            try (ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(bytes))) {
+                                result = ois.readObject();
+                            }
+                        }
+                    }
+                }
+            } catch (IOException | ClassNotFoundException e) {
+                throw new CacheProxyException("Error occurred while loading file: "
+                        + file.getName() + "from a zip", e);
+            }
+
+            return result;
+        }
+
+        private byte[] getBytes(InputStream in) throws IOException {
+            byte[] buffer = new byte[in.available()];
+            try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+                int r;
+
+                while ((r = in.read(buffer)) != -1) {
+                    baos.write(buffer,0, r);
+                }
+
+                return baos.toByteArray();
+            }
+        }
+
+        private void writeToCacheFile(File file, Object toSave) {
+            try {
+                isSerializableCheck(toSave);
+
+                if (isZip(file.getName())) {
+                    writeZipFile(file, toSave);
+                } else {
+                    writeToFile(file, toSave);
+                }
+            } catch (IOException e) {
+                throw new CacheProxyException("Error occurred while writing object: "
+                        + toSave + " to file: " + file.getName(), e);
+            }
+        }
+
+        private void writeZipFile(File file, Object toSave) throws IOException {
+            isSerializableCheck(toSave);
+
+            File zipFile = new File(file.getAbsolutePath() + ".zip");
+            try (ZipOutputStream zout = new ZipOutputStream(new FileOutputStream(zipFile));
+                 ByteArrayOutputStream baos = new ByteArrayOutputStream();
```

Comment:
&#39;try resource&#39; has incorrect indentation level 17, expected level should be 20.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1394172051>*